pinot-controller/src/main/java/org/apache/pinot/controller/recommender/data/writer/CsvWriter.java

```diff
@@ -38,6 +38,13 @@ protected String generateRow(DataGenerator generator) {
     return StringUtils.join(values, ",");
   }
 
+  @Override
+  protected void preprocess(java.io.FileWriter writer)
+      throws Exception {
```

Comment:
Why does this function throw exceptions?

Reply:
`preprocess(`) will work with the `FileWriter` object, here doing an `append()`, hence relaying the exception back so that it is handled by the caller.